### PR TITLE
chore: replace eslint with biome

### PR DESCRIPTION
**Pull request by aslilac**
Goodbye ESLint, and all the config and dependencies and waiting that came with it.

Instead, we can use Biome! Has all the lint rules that I think are "important" and is sooooo much faster.

### CODE REVIEW

**Comment by matifali** [comment]
To avoid the large diff can we keep the tab width to 2 spaces?

**Comment by aslilac** [comment]
I actually feel pretty strongly that we should take this opportunity to just go all in on the defaults of our code formatter. Every setting we change is a bit self defeating, when the purpose of the tool is to eliminate code style discussions.

I'd be ok with doing the tab transition in a separate PR to make this one more reviewable, but I think the plan should be to stick strongly with the defaults. As a bonus, we use tabs in our Go code because `go fmt` doesn't even have an option to use anything else, and this would be a nice bit of consistency to gain.

File: site/src/utils/filters.ts [hunk 1]
```diff
@@ -2,5 +2,5 @@ export function prepareQuery(query: string): string;
 export function prepareQuery(query: undefined): undefined;
 export function prepareQuery(query: string | undefined): string | undefined;
 export function prepareQuery(query?: string): string | undefined {
-  return query?.trim().replace(/  +/g, " ");
+  return query?.trim().replace(/ {2,}/g, " ");
```
Comment (johnstcn) [inline site/src/utils/filters.ts:5]: :eyes: it introspects regular expressions too? Nice.

File: site/src/pages/WorkspaceSettingsPage/WorkspaceSchedulePage/schedule.ts [hunk 2]
```diff
@@ -89,9 +89,8 @@ export const scheduleToAutostart = (schedule?: string): Autostart => {
       autostartEnabled: true,
       ...transformSchedule(schedule),
     };
-  } else {
-    return { autostartEnabled: false, ...emptySchedule };
   }
+  return { autostartEnabled: false, ...emptySchedule };
```
Comment (johnstcn) [inline site/src/pages/WorkspaceSettingsPage/WorkspaceSchedulePage/schedule.ts:93]: I'm a fan of this :+1:

**Review by johnstcn — COMMENTED** [review]
I smoke-tested this and didn't find any UI issues (but this was not an exhaustive test).

However, I did find a significant speedup from 54s for `eslint+prettier` to 10s for `biome`.

I'm holding back approval for the FE folks, but I like the overall idea of not needing to think about formatting. I'm also a fan of the 'auto-fix' decisions that Biome makes.

**Review by BrunoQuaresma — APPROVED** [review]
If the tests pass, I'm fine with these changes. I have a few questions, but I trust your judgment on when to merge them.

Questions:

- Do we need to make any changes on the Coder template side?
- Is there a VSCode plugin we should use, similar to what we did with Prettier? If so, it would be helpful to describe how this will affect our existing workflow and post a note about it in #dev.

File: Makefile [hunk 3]
```diff
@@ -401,22 +401,18 @@ fmt/go:
 	go run mvdan.cc/gofumpt@v0.4.0 -w -l .
 .PHONY: fmt/go
```
[the remark this answers is not in the document]
Comment (aslilac) [reply Makefile:403]: Actually, we still need prettier for basically everything it was doing before outside of site/, so I'll just put that job back.

**Comment by matifali** [comment]
@BrunoQuaresma We can add https://marketplace.visualstudio.com/items?itemName=biomejs.biome to .vscode/

File: .vscode/settings.json [hunk 4]
```diff
@@ -224,5 +223,16 @@
...
+  "[javascript][javascriptreact][json][jsonc][typescript][typescriptreact]": {
+    "editor.defaultFormatter": "biomejs.biome",
+    "editor.codeActionsOnSave":{
+      "source.organizeImports.biome": "explicit"
+    }
+  },
+
+  "[css][html][markdown]": {
+    "editor.defaultFormatter": "biomejs.biome"
+  }
```
Comment (matifali) [inline .vscode/settings.json:237]: it does not work with markdown right?

Comment (aslilac) [reply .vscode/settings.json:237]: whoops, good catch. I meant to put prettier for these.

**Comment by code-asher** [comment]
You can ignore whitespace changes in diffs (in GitHub too) so no harm to reviewability if we want to change the indentation now imo

**Comment by aslilac** [comment]
> You can ignore whitespace changes in diffs (in GitHub too) so no harm to reviewability if we want to change the indentation now imo

This PR was just already massive. 😅

**Pull request merged**